### PR TITLE
(PC-14846) feat(search): execute search with at least one letter

### PR DESCRIPTION
**Pull request by cedricls1**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-14846

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/search/components/__tests__/SearchBox.native.test.tsx [hunk 1]
```diff
@@ -76,4 +76,38 @@ describe('SearchBox component', () => {
...
     )
   })
+
+  it('should not execute a search if input is empty', () => {
+    const { getByPlaceholderText } = render(<SearchBox searchInputID={searchInputID} />)
+    const searchInput = getByPlaceholderText('Offre, artiste...')
+
+    fireEvent(searchInput, 'onSubmitEditing', { nativeEvent: { text: '' } })
+
+    expect(navigate).not.toBeCalledWith(
```
Comment (voisinhugo) [inline src/features/search/components/__tests__/SearchBox.native.test.tsx:86]: techniquement, on veut tester `not.toBeCalled()` plutôt que `not.toBeCalledWith(…)`

File: src/features/search/components/__tests__/SearchBoxRework.native.test.tsx [hunk 2]
```diff
@@ -133,4 +133,38 @@ describe('SearchBox component', () => {
...
     expect(searchInput.props.value).toBe('')
   })
+
+  it('should not execute a search if input is empty', () => {
+    const { getByPlaceholderText } = render(<SearchBoxRework searchInputID={searchInputID} />)
+    const searchInput = getByPlaceholderText('Offre, artiste...')
+
+    fireEvent(searchInput, 'onSubmitEditing', { nativeEvent: { text: '' } })
+
+    expect(navigate).not.toBeCalledWith(
```
Comment (voisinhugo) [inline src/features/search/components/__tests__/SearchBoxRework.native.test.tsx:143]: idem : techniquement, on veut tester `not.toBeCalled()` plutôt que `not.toBeCalledWith(…)`

File: src/ui/components/inputs/SearchInput.tsx [hunk 3]
```diff
@@ -83,7 +88,7 @@ const WithRefSearchInput: React.ForwardRefRenderFunction<RNTextInput, SearchInpu
         <BaseTextInput
           {...nativeProps}
           nativeID={searchInputID}
-          ref={forwardedRef}
+          ref={forwardedRef || searchInput}
```
Comment (voisinhugo) [inline src/ui/components/inputs/SearchInput.tsx:91]: Ça casse pas si on utilise une ref lors de l'utilisation de SearchInput ? Genre si on a une `forwardedRef`

File: src/features/search/components/__tests__/SearchBox.native.test.tsx (hunk 1 above)
Comment (cedricls1) [reply src/features/search/components/__tests__/SearchBox.native.test.tsx:86]: Ok simplifions :)

**Pull request merged**